### PR TITLE
feat: add SF_HEAP_REDEFINE_NEW to some structs,Leveled object decoding, change baseListCount to std::uint8 and other RE

### PR DESCRIPTION
**Pull request by TommInfinite**

### CODE REVIEW

File: CommonLibSF/include/RE/B/BGSMod.h [hunk 1]
```diff
@@ -31,7 +31,22 @@ namespace RE::BGSMod
...
+				struct Include
+				{
+					RE::BGSMod::Attachment::Mod* 					mod;
+					std::uint8_t                					level;
+					stl::enumeration<IncludesFlag, std::uint8_t> 	flags;
+				};
+				static_assert(sizeof(Include) == 0x10);
+
+				BSTArray<Include> includes;    // 00
 				BSTArray<void*> properties;  // 10
```
Comment (qudix) [inline CommonLibSF/include/RE/B/BGSMod.h:50]: Align names and comments

File: CommonLibSF/include/RE/B/BGSMod.h [hunk 2]
```diff
@@ -31,7 +31,22 @@ namespace RE::BGSMod
 			{
 			public:
 				// members
```
Comment (qudix) [inline CommonLibSF/include/RE/B/BGSMod.h:33]: Move this down to the members

File: CommonLibSF/include/RE/T/TESFile.h [hunk 3]
```diff
@@ -6,5 +6,9 @@ namespace RE
 	{
 	public:
 		SF_RTTI_VTABLE(TESFile);
+
+		std::uint8_t pad0[0x38];
+		char fileName[260];
+		//
```
[the remark this answers is not in the document]
Comment (TommInfinite) [reply CommonLibSF/include/RE/T/TESFile.h:12]: Aren't slashes supposed to say that  "there are more members below"?

File: CommonLibSF/include/RE/B/BGSMod.h (hunk 2 above)
Comment (TommInfinite) [reply CommonLibSF/include/RE/B/BGSMod.h:33]: Could you please rephrase? I don't understand.

File: CommonLibSF/include/RE/T/TESFile.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (qudix) [reply CommonLibSF/include/RE/T/TESFile.h:12]: I haven't seen that used anywhere before, but it doesn't really matter

File: CommonLibSF/include/RE/B/BGSMod.h (hunk 2 above)
Comment (qudix) [reply CommonLibSF/include/RE/B/BGSMod.h:33]: Move the comment to be right above the class members at the bottom

**Pull request merged**